tools/apigee-envoy-quickstart/aekitctl.sh

```diff
@@ -0,0 +1,150 @@
+        "-a --action      : Install or Delete action, valid values 'install', 'delete'\n" \
+        "-p --platform    : Standalone install for platform opdk or edge, valid values 'opdk', 'edge'\n"
+    exit 1
+}
+
+init() {
+    export CLUSTER_CTX="gke_${PROJECT_ID}_${CLUSTER_LOCATION}_${CLUSTER_NAME}"
+    export ENVOY_AX_SA="x-apigee-envoy-adapter-2-sa"
+    export CLI_HOME=$ENVOY_HOME/apigee-remote-service-cli
+    export REMOTE_SERVICE_HOME=$ENVOY_HOME/apigee-remote-service-envoy
+    export ENVOY_CONFIGS_HOME=$CLI_HOME/envoy-configs-and-samples
+    export AX_SERVICE_ACCOUNT=$ENVOY_HOME/$ENVOY_AX_SA.json
+    export NAMESPACE="apigee"
+    if [ "$PLATFORM" == 'edge' ]; then
+        export MGMT_HOST="https://api.enterprise.apigee.com"
+    fi
+    if [[ -z $PROJECT_ID ]] && [[ $INSTALL_TYPE == 'standalone-apigee-envoy' ]]; then
+        export PROJECT_ID=$APIGEE_PROJECT_ID;
+    fi
+}
+
+createDir() {
+    mkdir "$CLI_HOME"
+    mkdir "$REMOTE_SERVICE_HOME"
+}
+
+PARAMETERS=()
+
+if [[ $# -ne 4 && $# -ne 6 ]]
+then
+   usage;
+   exit 1;
+fi  
+
+while [[ $# -gt 0 ]]
+do
+    param="$1"
+    case $param in
+        -t|--type)
+        export INSTALL_TYPE="$2"
+        shift
+        shift
+        ;;
+        -a|--action)
+        export ACTION="$2"
+        shift
+        shift
+        ;;
+        -p|--platform)
+        export PLATFORM="$2"
+        shift
+        shift
+        ;;
+        *)
+        PARAMETERS+=("$1")
+        shift
+        ;;
+    esac
+done
+ 
+if [[ -z $INSTALL_TYPE ]]; then
+    usage "installation type is a mandatory field"
+fi
+
+if [[ -z $ACTION ]]; then
+    usage "action is a mandatory field"
+fi
+
+if [ "$PLATFORM" != 'opdk' ] && [ "$PLATFORM" != 'edge' ]
+then
+    gke-gcloud-auth-plugin --version > /dev/null 2>&1
+    RESULT=$?
+    if [ $RESULT -eq 0 ]; then
+        export USE_GKE_GCLOUD_AUTH_PLUGIN=True
+    fi
+fi
+
+init;
+
+if [ "$PLATFORM" == 'opdk' ] || [ "$PLATFORM" == 'edge' ]
+then
+    TOKEN=$(echo -n "$APIGEE_USER":"$APIGEE_PASS" | base64 | tr -d \\r)
+    export TOKEN;
+    export TOKEN_TYPE="Basic"
+else
+    export TOKEN_TYPE="Bearer"
+    export MGMT_HOST="https://apigee.googleapis.com"
+fi
+
+./scripts/validate.sh
+
+if [ "$PLATFORM" == 'opdk' ] || [ "$PLATFORM" == 'edge' ]
+then
+    ./scripts/validate-opdk-edge-setup.sh
+else
+    ./scripts/validate-new-gen-setup.sh
+    export APIGEE_ORG=$APIGEE_X_ORG
+    export APIGEE_ENV=$APIGEE_X_ENV
+fi
+
+if [[ "$INSTALL_TYPE" == 'istio-apigee-envoy' ]] && [[ "$ACTION" == 'install' ]]; then
+    createDir;
+    echo "Installing istio-apigee-envoy"
+    export TEMPLATE=$ISTIO_TEMPLATE_VER
+    ./istio-apigee-envoy-install.sh
+elif [[ "$INSTALL_TYPE" == 'istio-apigee-envoy' ]] && [[ "$ACTION" == 'delete' ]]; then
+    echo "Deleting istio-apigee-envoy"
+    ./scripts/delete-apigee-envoy-setup.sh
+elif [ "$INSTALL_TYPE" == 'standalone-apigee-envoy' -a "$ACTION" == 'install' ]; then
```

Comment:
Shellcheck highlighted this because of [SC2166](https://www.shellcheck.net/wiki/SC2166)

A good example can be found in here: https://github.com/git/git/blob/v2.6.0/Documentation/CodingGuidelines#L154